GWXtreme/parametrized_eos_sampler.py

```diff
@@ -179,7 +179,61 @@ def save_data(self):
         f.create_dataset('logp',data=np.array(self.logp))
         f.close()
         
+    def parse_samples(self,burn_in_frac = 0.5, thinning = None):
+        '''
+        This methods loads a .h5 file with the MCMC samples 
+        of EoS hyper-parameters and parses them for use by the 
+        plotting functions.
+        see https://emcee.readthedocs.io/en/stable/tutorials/autocorr/
+        for some documentation on choosing thinning and burn-in
+        
+        burn_in_frac  :: fraction of samples to discard from each chain
+                         for MCMC burn in. Default corresponds to discarding 
+                         half the samples in each chain.
+        
+        thinning      :: Number of samples to skip in each chain post
+                         burn in. "None" implements the default value 
+                         which is either (length of chain)/50 or half of
+                         the maximum integrated autocorrelation time. The 
+                         former is used in case autocorrelation analysis 
+                         yields throws non-convergence error. For no thinning 
+                         set thinning=1
+        
+        '''
```

Comment:
Provide example of the implementation

Reply:
How about this?

```
Stand-alone example (once the sampler class is initialized and the samples loaded,):
>>> sampler.parse_samples()
>>> samples = sampler.samples
```

Reply:
Maybe one more line before that instantiating the `sampler` object.

Reply:
Like this then?
```
Stand-alone example:
>>> sampler_spectral=mcmc_sampler([],  
        {'gamma1':{'params':{"min":0.2,"max":2.00}},
        'gamma2':{'params':{"min":-1.6,"max":1.7}},
        'gamma3':{'params':{"min":-0.6,"max":0.6}},
        'gamma4':{'params':{"min":-0.02,"max":0.02}}},
        out, nwalkers=100, Nsamples=10000, ndim=4,
        spectral=True,npool=16)
>>> sampler.parse_samples()
>>> samples = sampler.samples
```

Reply:
Also write some dummy posterior sample file names instead of and empty list.

Reply:
If posterior files are provided it will attempt to stack them and the dummy paths will lead to an error. When instantiated for post-processing the argument is supposed to be an empty list. Do you still want me to add that to the doc string ? It might be confusing to the users.

Reply:
I'm not sure I follow this. This is supposed to be a standalone method that will take as input the file with the gamma posteriors, not the Bilby posteriors. When we run this `parse_samples` method it should be running on an existing GWXtreme result. So, then why are we using the Bilby posterior file again (that's what I gather from your previous statement). Isn't it supposed to read any gamma posterior and produce the plots?

Reply:
The point is this stand-alone method is a part of the mcmc_sampler class. The first argument required to instantiate this class is a list of bilby posterior files because most commonly this class is expected to be initialized for the purpose of EoS inference. However, If a particular function from this class is to be called in postprocessing after EoS inference is done then the same class needs to be instantiated as well but in this case without any bilby posterior samples since the EoS inference is already done. However, instantiating the class itself requires the first argument which is why an empty list is provided there. The empty list prevents the class from attempting to stack single event EoS models since that is not needed in post-processing. The correct way to call this function is thus:

```
>>> sampler_spectral=mcmc_sampler([],  
        {'gamma1':{'params':{"min":0.2,"max":2.00}},
        'gamma2':{'params':{"min":-1.6,"max":1.7}},
        'gamma3':{'params':{"min":-0.6,"max":0.6}},
        'gamma4':{'params':{"min":-0.02,"max":0.02}}},
        out, nwalkers=100, Nsamples=10000, ndim=4,
        spectral=True,npool=16)
>>> sampler.load_samples('file/containing/EoS/hyperparameter/samples')
>>> sampler.parse_samples()
>>> samples = sampler.samples
>>>
```

The ```load_samples``` was missing in my previous comments, sorry about that.

The way I had designed this class originally, was that after EoS inference is finished, samples could be parsed and plots made as part of the same job/program/driver that does the EoS inference. Which is why these two functions were not there explicitly. However, the reviewer requested these be there in case someone needs to re-produce the plots after the EoS inference has finished as part of a separate job or program in a stand alone fashion. The way to do that is the code block above which is what I thought should go to the doc string. 

Does this make sense?